diff --git a/types/inferrawdoctype.d.ts b/types/inferrawdoctype.d.ts
index 59c3d7caf2c4cb2d0fc15ec227d67a0098f91a1e..e94c980c6e3e96a1f9292e63c05debd8cafddc64 100644
--- a/types/inferrawdoctype.d.ts
+++ b/types/inferrawdoctype.d.ts
@@ -72,7 +72,7 @@ declare module 'mongoose' {
         : // If the type key isn't callable, then this is an array of objects, in which case
           // we need to call InferRawDocType to correctly infer its type.
           Array<InferRawDocType<Item>>
-      : IsSchemaTypeFromBuiltinClass<Item> extends true ? ResolveRawPathType<Item, { enum: Options['enum'] }, TypeKey>[]
+      : IsSchemaTypeFromBuiltinClass<Item> extends true ? ResolveRawPathType<Item, {}, TypeKey>[]
       : IsItRecordAndNotAny<Item> extends true ?
         Item extends Record<string, never> ?
           ObtainRawDocumentPathType<Item, TypeKey>[]
diff --git a/types/inferschematype.d.ts b/types/inferschematype.d.ts
index 7423f40021512c0d4f4ac18445ab2c4a4672a7b3..a9f919dbdd5f90adbe94f6923a57253a7ba383cc 100644
--- a/types/inferschematype.d.ts
+++ b/types/inferschematype.d.ts
@@ -282,7 +282,7 @@ type ResolvePathType<
         : // If the type key isn't callable, then this is an array of objects, in which case
           // we need to call ObtainDocumentType to correctly infer its type.
           Types.DocumentArray<ObtainDocumentType<Item, any, { typeKey: TypeKey }>>
-      : IsSchemaTypeFromBuiltinClass<Item> extends true ? ResolvePathType<Item, { enum: Options['enum'] }, TypeKey>[]
+      : IsSchemaTypeFromBuiltinClass<Item> extends true ? ResolvePathType<Item, {}, TypeKey>[]
       : IsItRecordAndNotAny<Item> extends true ?
         Item extends Record<string, never> ?
           ObtainDocumentPathType<Item, TypeKey>[]
